### PR TITLE
Add better options management

### LINKED ISSUE #141: [IDEA] Better options handling
**What is your feature about ?**
We manage by hand all the option given when a filter is launched.
It can be a good idea to use a lib for that.

**Describe what you'd like**
Use a lib for option managing

**Additional context**
We can use getopt. There is a Boost solution also.
Proposed by @KGuillemot

### PR DESCRIPTION
**Pull request by NS4nti**
# :sparkles: Pull Request Template
:bangbang: Once all the **checklist** is **done** you have to:
  * **stash merge** this pull request
  * **delete** the corresponding **branch**
  * **close** the associated **issue**

## :page_with_curl: Type of change

**Breaking change**: fix or feature that would cause existing functionality to not work as expected.

## :bulb: Related Issue(s)

- Resolve #141 

## :black_nib: Description

Now the option are handled by getopt. Also they are more conform : 
`./darwin [-h] [-l [DEBUG|INFO|WARNING|ERROR|CRITICAL]] filter_name socket_path config_file monitoring_socket_path pid_file output next_filter_socket_path nb_thread cache_size threshold`
### FreeBSD 12.0
- clang++  6.0.1
- CMake  3.7.2
- Python  3

### Ubuntu 19.0.4
- g++  8.3.0
- CMake  3.7.2
- Python  3

## :heavy_check_mark: Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

</br>

- [x] :raising_hand: **I certify on my honor that all the information provided is true, and I've done all I can to deliver a high quality code**

### CODE REVIEW

File: samples/base/Core.cpp [hunk 1]
```diff
@@ -146,15 +181,10 @@ namespace darwin {
...
-                << "  -z\tDeveloper mode, does not create a daemon, set log level to debug"
-                << std::endl;
-        std::cout << "  -d\tSet log level to debug" << std::endl;
-        std::cout << "  -i\tSet log level to info" << std::endl;
-        std::cout << "  -n\tSet log level to notice" << std::endl;
-        std::cout << "  -w\tSet log level to warning (DEFAULT)" << std::endl;
-        std::cout << "  -e\tSet log level to error" << std::endl;
-        std::cout << "  -c\tSet log level to critical" << std::endl;
+        std::cout << "  -l\tSet log level to [DEBUG|INFO|NOTICE|WARNING|ERROR|CRITICAL|DEVELOPER].\n" 
+                  << "    \tDefault it's WARNING, DEVELOPER mode does not create a daemon and log level is DEBUG."
```
Comment (HugoSoszynski) [inline samples/base/Core.cpp]:
```suggestion
                  << "    \tDefault is WARNING."
                  << "    \tNOTE: DEVELOPER mode does not create a daemon and sets log level to DEBUG."
```

File: samples/base/Logger.hpp [hunk 2]
```diff
@@ -86,6 +89,12 @@ namespace darwin {
             /// \param type The log level to set.
             void setLevel(log_type type);
 
+            /// Set the log Level with a given string.
+            ///
+            /// \param level The string that describe the log level wanted. 
+            /// \return true if success, false otherwise.
+            bool setLogLevel(std::string level);
```
Comment (HugoSoszynski) [inline samples/base/Logger.hpp]:
```suggestion
            bool setLevel(std::string level);
```
Why not use polymorphism ?
Adding multiple name for the same kind of functionality may be confusing.

File: tests/tools/filter.py [hunk 3]
```diff
@@ -8,16 +8,16 @@
...
         self.socket = socket_path if socket_path else "/tmp/{}.sock".format(filter_name)
         self.config = config_file if config_file else "/tmp/{}.conf".format(filter_name)
         self.path = path if path else "{}darwin_{}".format(DEFAULT_FILTER_PATH, filter_name)
         self.monitor = monitoring_socket_path if monitoring_socket_path else "/tmp/{}_mon.sock".format(filter_name)
         self.pid = pid_file if pid_file else "/tmp/{}.pid".format(filter_name)
-        self.cmd = [self.path, self.filter_name, self.socket, self.config, self.monitor, self.pid, output, next_filter_socket_path, str(nb_thread), str(cache_size), str(thresold), log_level]
+        self.cmd = [self.path, "-l", log_level, self.filter_name, self.socket, self.config, self.monitor, self.pid, output, next_filter_socket_path, str(nb_thread), str(cache_size), str(thresold)]
         self.process = None
-        self.error_code = 99 # For valgrind testing
+        self.error_code = 99  # For valgrind testing
```
Comment (HugoSoszynski) [inline tests/tools/filter.py]:
```suggestion
        self.error_code = 99 # For valgrind testing
```

File: samples/base/Logger.hpp (hunk 2 above)
Comment (NS4nti) [reply samples/base/Logger.hpp]: Done

File: samples/base/Core.cpp [hunk 4]
```diff
@@ -75,47 +78,79 @@ namespace darwin {
...
+                case '?':  
+                    DARWIN_LOG_CRITICAL("Core:: Program Arguments:: Unknown option");
+                    Core::Usage();
+                    return false;
+            }  
+        }  
+
+        // After options we need 10 mandatory arguments
+        if(ac-optind < 10){
             DARWIN_LOG_CRITICAL("Core:: Program Arguments:: Missing some parameters");
```
Comment (frikilax) [inline samples/base/Core.cpp]:
```suggestion
            DARWIN_LOG_ERROR("Core:: Program Arguments:: Missing some parameters");
```

**Pull request merged**